src/Mod/Sketcher/Gui/DrawSketchHandlerBSplineByInterpolation.h

```diff
@@ -0,0 +1,606 @@
+                    BSplineKnots.pop_back();
+                    sugConstr.pop_back();
+
+                    return true;
+                }
+            }
+
+            // insert point for knot, defer internal alignment constraining.
+            try {
+
+
+                //Add knot
+                Gui::cmdAppObjectArgs(sketchgui->getObject(), "addGeometry(Part.Point(App.Vector(%f,%f,0)),True)",
+                                      BSplineKnots.back().x,BSplineKnots.back().y);
+
+                knotGeoIds.push_back(getHighestCurveIndex());
+            }
+            catch (const Base::Exception& e) {
+                Base::Console().Error("%s\n", e.what());
+                Gui::Command::abortCommand();
+
+                static_cast<Sketcher::SketchObject *>(sketchgui->getObject())->solve();
+
+                return false;
+            }
+
+            // add auto constraints on knot
+            if (!sugConstr.back().empty()) {
+                createAutoConstraints(sugConstr.back(), knotGeoIds.back(), Sketcher::PointPos::start, false);
+            }
+
+            if (!IsClosed) {
+                addSugConstraint();
+            }
+
+        }
+        return true;
+    }
+
+    bool releaseButton(Base::Vector2d onSketchPos) override
+    {
+        prevCursorPosition = onSketchPos;
+        MousePressMode = MOUSE_NOT_PRESSED;
+
+        return finishCommand(onSketchPos);
+    }
+
+    void registerPressedKey(bool pressed, int key) override
+    {
+        // if (SoKeyboardEvent::D == key && pressed) {
+        //     SplineDegree = QInputDialog::getInt(
+        //         Gui::getMainWindow(),
+        //         QObject::tr("B-Spline Degree"),
+        //         QObject::tr("NOTE: For construction by interpolation, this degree is ignored for now.\n"
+        //                     "Define B-Spline Degree, between 1 and %1:")
+        //         .arg(QString::number(Geom_BSplineCurve::MaxDegree())),
+        //         SplineDegree, 1, Geom_BSplineCurve::MaxDegree(), 1);
+        //     // FIXME: Pressing Esc here also finishes the B-Spline creation.
+        //     // The user may only want to exit the dialog.
+        // }
+        if (SoKeyboardEvent::M == key && pressed) {
+            // TODO: On pressing, say, M, modify the knot's multiplicity
+            if (BSplineMults.size() > 1) {
+                BSplineMults.back() = QInputDialog::getInt(
+                    Gui::getMainWindow(),
+                    QObject::tr("Set knot multiplicity"),
+                    QObject::tr("NOTE 1: For construction by interpolation, the de facto maximum is currently 3.\n"
+                                "NOTE 2: Under certain circumstances (details WIP), this value may be ignored.\n"
+                                "Set knot multiplicity at the last point provided, between 1 and %1:")
+                    .arg(QString::number(SplineDegree)),

```

Comment:
Consider if you really want these notes here.

Perhaps these notes are part of the documentation of the tool. The user can do nothing about it. In other words, the user will have to enter 1-3, because 3 is our degree and we cannot go beyond. So he will enter a number 1-3.  The explanations do not quite help the choice IMO.

If the user does not understand the tool, then the user will go to the documentation. There it can be explained more in detail that the maximum is the degree of the BSpline and that any change to the multiplicity of endpoints will be ignored. The latter is actually what most users would expect.

Reply:
Oh! that is interesting. Where does it happen other than endpoints? (as this could help looking at a better message)

Ok, let's try to look into alternatives:
"Set knot multiplicity at the last entered point, between 1 and %1."
"Note that multiplicity will be ignored for endpoints, which will have %1 multiplicity"

Reply:
Please see https://github.com/FreeCAD/FreeCAD/pull/8530#issuecomment-1460289121 for examples. For periodic B-splines there is also an added complication that the first knot is multiplicity 2 (unnecessary in my opinion, but implemented in OCCT). Lots to test here. Something like "Note that multiplicity may be ignored under certain circumstances. Please refer to documentation for details".

Reply:
Ok. I agree.